### PR TITLE
Added new links and title display

### PR DESCRIPTION
**Pull request by Mi-Lan**

### CODE REVIEW

File: src/library/utils/index.ts [hunk 1]
```diff
@@ -47,7 +47,8 @@ export function shortenAddress(address?: string, charsBefore = 4, charsAfter = 4
  * @returns
  */
 export function getNimiLinkLabel(nimi: NimiLinkBaseDetails): string {
-  if (nimi.type === NimiLinkType.URL && isValidUrl(nimi.content)) {
+  if (nimi.title) return nimi.title;
```
Comment (Mi-Lan) [inline src/library/utils/index.ts:50]: @adamazad Any suggestion how to handle the link tittle better?

File: src/library/assets/svg/links/snapchat.svg [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
+<path fill-rule="evenodd" clip-rule="evenodd" d="M10 -6.79868e-08C6.68512 -4.54498e-08 3.99787 2.68629 3.99787 6L3.99787 7.875L3.09755 7.2C2.65556 6.86863 2.02854 6.95817 1.69705 7.4C1.36556 7.84183 1.45513 8.4686 1.89712 8.8L3.7643 10.1999C3.31957 11.3676 2.23602 12.5905 0.482125 13.6425C-0.000707009 13.9321 -0.147879 14.5864 0.164452 15.0547C1.28274 16.7315 3.05738 18 4.99822 18C6.78322 18 8.00684 20 10 20C12.0121 20 13.2103 18 15.0017 18C16.9425 18 18.7174 16.7316 19.8356 15.0547C20.1479 14.5863 20.0007 13.9321 19.5179 13.6425C17.764 12.5905 16.6804 11.3676 16.2357 10.1999L18.1029 8.8C18.5449 8.4686 18.6344 7.84183 18.303 7.4C17.9715 6.95817 17.3445 6.86863 16.9025 7.2L16.0021 7.87504L16.0022 6.00003C16.0022 2.68632 13.3149 -9.05236e-08 10 -6.79868e-08Z" fill="#FFC700"/>
+</svg>
```
Comment (adamazad) [inline src/library/assets/svg/links/snapchat.svg:3]: Where did we get the logos from? Snapshot logo doesn't look crisp

File: src/library/types/NimiLink.ts [hunk 3]
```diff
@@ -51,6 +51,38 @@ export enum NimiLinkType {
...
   DISCORD = 'DISCORD',
+  /**
+   * WhatsApp Profile
+   */
+  WHATSAPP = 'WHATSAPP',
+  /**
+   * Messanger Profile
+   */
+  MESSANGER = 'MESSANGER',
+  /**
```
Comment (adamazad) [inline src/library/types/NimiLink.ts:62]: Typo 
```suggestion
  /**
   * Messenger Profile
   */
  MESSENGER = 'MESSENGER',
  /**
```

File: src/library/types/NimiLink.ts [hunk 4]
```diff
@@ -51,6 +51,38 @@ export enum NimiLinkType {
...
+   */
+  SNAPCHAT = 'SNAPCHAT',
+  /**
+   * Facebook Profile
+   */
+  FACEBOOK = 'FACEBOOK',
+  /**
+   * Dribble Profile
+   */
+  DRIBBLE = 'DRIBBLE',
```
Comment (adamazad) [inline src/library/types/NimiLink.ts:81]: Typo 
```suggestion
  DRIBBBLE = 'DRIBBBLE',
```

File: public/data.json [hunk 5]
```diff
@@ -7,12 +7,13 @@
...
-    { "type": "DISCORD", "label": "Discord", "content": "violet#6640" },
-    { "type": "EMAIL", "label": "Email", "content": "smilenostress@gmail.com" },
-    { "type": "URL", "label": "Website", "content": "" },
-    { "type": "TWITTER", "label": "Twitter", "content": "0xViolet" },
-    { "type": "LINKEDIN", "label": "Github", "content": "https://linkedin.com/in/Mi-lan" },
-    { "type": "GITHUB", "label": "Github", "content": "https://github.com/Mi-lan" },
-    { "type": "LENSTER", "label": "Lenster", "content": "luduvigo.lens" }
+    { "type": "DISCORD", "title": "", "content": "violet#6640" },
+    { "type": "EMAIL", "title": "Email", "content": "smilenostress@gmail.com" },
+    { "type": "URL", "title": "Website", "content": "" },
```
Comment (adamazad) [inline public/data.json:12]: Add a URL here

File: src/library/assets/svg/links/dribble.svg [hunk 6]
```diff
@@ -0,0 +1,3 @@
...
+</svg>
```
Comment (adamazad) [inline src/library/assets/svg/links/dribble.svg:3]: Filename typoe

File: src/library/constants/index.tsx [hunk 7]
```diff
@@ -7,14 +7,26 @@ import { ReactComponent as LitecoinLogoUrl } from '../assets/svg/blockchain/lite
 import { ReactComponent as PolygonLogoUrl } from '../assets/svg/blockchain/polygon.svg';
 import { ReactComponent as SolanaLogoUrl } from '../assets/svg/blockchain/solana.svg';
 import { ReactComponent as DiscordLogo } from '../assets/svg/links/discord.svg';
+import { ReactComponent as DribbleLogo } from '../assets/svg/links/dribble.svg';
```
Comment (adamazad) [inline src/library/constants/index.tsx:10]: Typo

File: src/library/utils/index.ts (hunk 1 above)
Comment (adamazad) [reply src/library/utils/index.ts:50]: I believe this is good. `title` property has the highest precedence.

**Review by adamazad — CHANGES_REQUESTED** [review]

File: src/library/assets/svg/links/snapchat.svg (hunk 2 above)
Comment (Mi-Lan) [reply src/library/assets/svg/links/snapchat.svg:3]: you mean snapchat...yes its from figma maybe zett intended it like that https://www.figma.com/file/IhAXgag9wFFBLLCW7pkv6i/Nimi-%7C-Devconnect-AMS-2022?node-id=1436%3A41065

File: public/data.json [hunk 8]
```diff
@@ -7,12 +7,14 @@
...
-    { "type": "GITHUB", "label": "Github", "content": "https://github.com/Mi-lan" },
-    { "type": "LENSTER", "label": "Lenster", "content": "luduvigo.lens" }
+    { "type": "DISCORD", "title": "", "content": "violet#6640" },
+    { "type": "EMAIL", "title": "Email", "content": "smilenostress@gmail.com" },
+    { "type": "URL", "title": "My website", "content": "https://linkedin.com/in/Mi-lan" },
+    { "type": "TWITTER", "title": "Twitter", "content": "0xViolet" },
+    { "type": "LINKEDIN", "title": "Github", "content": "https://linkedin.com/in/Mi-lan" },
+    { "type": "GITHUB", "title": "Github", "content": "https://github.com/Mi-lan" },
+    { "type": "LENSTER", "title": "Lenster", "content": "luduvigo.lens" },
+    { "type": "FIGMA", "title": "Lenster", "content": "luduvigo.lens" },
```
Comment (adamazad) [inline public/data.json:17]:
```suggestion
    { "type": "FIGMA", "title": "Figma", "content": "luduvigo" },
```

?

File: src/library/app/index.tsx [hunk 9]
```diff
@@ -12,7 +12,9 @@ export interface CardAppProps {
  * Providers, necessary data, and rendering.
  */
 export function CardApp({ nimi }: CardAppProps) {
-  const filteredNimi = filterEmptyLinks(nimi);
+  console.log('OG', nimi);
+  const filteredNimi = filterEmptyFields(nimi);
+  console.log('filtered', filterEmptyFields);
```
Comment (adamazad) [inline src/library/app/index.tsx:17]: You can wrap the console statement in

`process.env.NODE_ENV !== 'development'`

File: src/library/utils/index.ts [hunk 10]
```diff
@@ -76,9 +77,10 @@ export function generateLink({ type, content }: NimiLinkBaseDetails): string {
  * @param nimi - Nimi object
  * @returns Filtered nimi object
  */
-export function filterEmptyLinks(nimi: Nimi): Nimi {
+export function filterEmptyFields(nimi: Nimi): Nimi {
   if (nimi.links) nimi.links = nimi.links.filter(({ content }) => content !== '');
-
+  if (nimi.addresses) nimi.addresses = nimi.addresses.filter(({ address }) => address !== '');
+  console.log('modified', nimi);
```
Comment (adamazad) [inline src/library/utils/index.ts:83]: We should be lean with console statements

**Review by adamazad — APPROVED** [review]
Address the comments and merge. Good work!

**Pull request merged**